### PR TITLE
Allow creating apps with version in unit tests and integration tests

### PR DESCRIPTION
**Pull request by gokulavasan**
Summary:
This will allow us to write end-to-end unit tests and integration tests for application versioning.

Build : http://builds.cask.co/browse/CDAP-RUT189

### CODE REVIEW

File: cdap-test/src/main/java/co/cask/cdap/test/TestManager.java [hunk 1]
```diff
@@ -79,6 +79,16 @@ ApplicationManager deployApplication(Id.Namespace namespace, Class<? extends App
   ApplicationManager deployApplication(Id.Application appId, AppRequest appRequest) throws Exception;
 
   /**
+   * Deploys an {@link Application}.
```
Comment (ecfm) [inline cdap-test/src/main/java/co/cask/cdap/test/TestManager.java]: Creates an application with a version using an existing artifact?

**Comment by gokulavasan** [comment]
@maochf Addressed comment. Please take a look again.

**Comment by gokulavasan** [comment]
@maochf PR is ready for review. Please take a look when you get a chance. Thanks!

File: cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java [hunk 2]
```diff
@@ -403,6 +406,37 @@ public void testAppFromArtifact() throws Exception {
...
+      new ArtifactSummary(artifactId.getArtifact(), artifactId.getVersion()),
+      new ConfigTestApp.ConfigClass("tS2", "tD2", "tV2"));
+    appManager = deployApplication(appId, createRequest);
+    serviceManager = appManager.getServiceManager(ConfigTestApp.SERVICE_NAME);
+    serviceManager.start();
+
+    serviceURL = serviceManager.getServiceURL();
+    Assert.assertEquals("tV2", gson.fromJson(callServiceGet(serviceURL, "ping"), String.class));
+    serviceManager.stop();
+  }
```
Comment (ecfm) [inline cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java:438]: is it necessary to delete apps at the end?

Comment (gokulavasan) [reply cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java:438]: Not necessary, since it is taken care of in the `@After` method.

**Comment by ecfm** [comment]
LGTM

**Pull request merged**